src/lib/component/logger.hpp

```diff
@@ -99,9 +103,9 @@ namespace details {
         return make_stream(std::string(s), t);
     }
     //! @brief Makes a stream reference from a stream pointer.
-    template <typename S, typename T>
-    std::shared_ptr<std::ostream> make_stream(std::ostream* o, const common::tagged_tuple<S,T>&) {
-        return std::shared_ptr<std::ostream>(o, [] (void*) {});
+    template <typename O, typename S, typename T>
+    std::shared_ptr<O> make_stream(O* o, const common::tagged_tuple<S,T>&) {
```

Comment:
dubbio già sul vecchio codice: perché `make_stream` ha un parametro `tagged_tuple` inutilizzato?

Reply:
> dubbio già sul vecchio codice: perché `make_stream` ha un parametro `tagged_tuple` inutilizzato?

It is not unused in every overload. The version with a `const std::string& s` first argument uses it, in case `s` is a directory, to automatically synthesize a file name from the values in the `tagged_tuple`